tests/components/knx/test_init.py

```diff
@@ -262,6 +268,61 @@ async def test_init_connection_handling(
             )
 
 
+@pytest.mark.parametrize(
+    ("state_updater_config"),
+    [True, False],
+)
+async def test_default_state_updater(
+    hass: HomeAssistant,
+    knx: KNXTestKit,
+    create_ui_entity: KnxEntityGenerator,
+    freezer: FrozenDateTimeFactory,
+    state_updater_config: KNXConfigEntryData,
+) -> None:
+    """Test default state updater is applied to xknx device instances."""
+    config_entry = MockConfigEntry(
+        title="KNX",
+        domain=KNX_DOMAIN,
+        data={
+            **DEFAULT_ENTRY_DATA,
+            CONF_KNX_CONNECTION_TYPE: CONF_KNX_AUTOMATIC,  # missing in default data
+            CONF_KNX_STATE_UPDATER: state_updater_config,
+        },
+    )
+    knx.mock_config_entry = config_entry
+    await knx.setup_integration({})
+    await create_ui_entity(
+        platform=Platform.SWITCH,
+        knx_data={
+            "ga_switch": {"write": "1/1/1", "state": "2/2/2"},
+            "respond_to_read": True,
+            "sync_state": True,  # True uses xknx default state updater
+            "invert": False,
+        },
+    )
+    # created entity sends read-request to KNX bus on connection
+    await knx.assert_read("2/2/2")
+    await knx.receive_response("2/2/2", True)
+
+    freezer.tick(timedelta(minutes=59))
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
+    await knx.assert_no_telegram()
+
+    freezer.tick(timedelta(minutes=1))  # 60 minutes passed
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
+    if state_updater_config:
+        # state updater reads again after 60 minutes
+        await knx.assert_read("2/2/2")
+        await knx.receive_response("2/2/2", True)
+    else:
+        # state updater does not read again
+        await knx.assert_no_telegram()
```

Comment:
Please don't use if statements in test code

Reply:
an alternative to https://github.com/home-assistant/core/pull/135611/commits/b5a4baa6173d65bc59a996520583fe3d32cd7cd5 would be something like 
```py
@pytest.mark.parametrize(
    ("state_updater_config", "final_assertion"),
    [
        (
            True,
            lambda knx: asyncio.gather(
                knx.assert_read("2/2/2"),
                knx.receive_response("2/2/2", True),
            ),
        ),
        (
            False,
            lambda knx: knx.assert_no_telegram(),
        ),
    ],
)
async def test_default_state_updater(..., final_assertion):
  ...
  await final_assertion(knx)
```
let me know what you prefer.